test/weights.jl

```diff
@@ -36,6 +36,23 @@ weight_funcs = (weights, aweights, fweights, pweights)
     @test sum(sa, wv) === 7.0
 end
 
+@testset "$f, setindex!" for f in weight_funcs
+    w = [1., 2., 3.]
+    wv = f(w)
+
+    @test wv[1] == 1.
+    sum(wv) == 6.
```

Comment:
Missing `@test`. Also if you use trailing dots, use `===`, else it has no effect and it's misleading.